pkg/ccl/importccl/import_stmt.go

```diff
@@ -1435,6 +1435,14 @@ func (r *importResumer) Resume(ctx context.Context, execCtx interface{}) error {
 		// Now that we know all the tables are offline, pick a walltime at which we
 		// will write.
 		details.Walltime = p.ExecCfg().Clock.Now().WallTime
+		// Update the job details as soon as we've chosen a timestamp. We don't wait
+		// until below to be defensive against hard-to-reason state where the
+		// details' timestamp is not set.
+		// We have seen the scan below hang in some states, which may cause and
+		// import failure.
+		if err := r.job.WithTxn(nil).SetDetails(ctx, details); err != nil {
```

Comment:
I don't think we need this one considering the change to when we revert. If we happen to error-out during WasEmpty checks, we actually would prefer, with your other change, to fast-skip the reverts since we never ingested?

Reply:
Agreed. Sitting on it longer, it seems like this would actually be worse, as you said. Removed this commit.